controllers/main.go

```diff
@@ -49,34 +49,41 @@ const (
 	agendasCacheLife = time.Hour
 )
 
+// Config struct to be passed as argument into NewMainController func
+// Reduces the number of marameters passed to NewMainController()
+type MainControllerParams struct {
+	AdminIPs        []string
+	AdminUserIDs    []string
+	APISecret       string
+	BaseURL         string
+	ClosePool       bool
+	ClosePoolMsg    string
+	PoolEmail       string
+	PoolFees        float64
+	PoolLink        string
+	RealIPHeader    string
+	MaxVotedTickets int
+	Description     string
+	Designation     string
+}
+
 // MainController is the wallet RPC controller type.  Its methods include the
 // route handlers.
 type MainController struct {
 	// embed type for c.Env[""] context and ExecuteTemplate helpers
 	system.Controller
 
-	adminIPs             []string
-	adminUserIDs         []string
-	APISecret            string
 	APIVersionsSupported []int
-	baseURL              string
-	closePool            bool
-	closePoolMsg         string
 	enableStakepoold     bool
 	feeXpub              *hdkeychain.ExtendedKey
 	StakepooldServers    *stakepooldclient.StakepooldManager
-	poolEmail            string
-	poolFees             float64
-	poolLink             string
 	params               *chaincfg.Params
-	realIPHeader         string
 	captchaHandler       *CaptchaHandler
 	emailSender          email.Sender
 	voteVersion          uint32
 	votingXpub           *hdkeychain.ExtendedKey
-	maxVotedTickets      int
-	description          string
-	designation          string
+
+	mainControllerParams MainControllerParams
```

Comment:
Lets rename this field `params`.

Calling it `mainControllerParams` means we need to write code like `mainController.mainControllerParams` when we could be writing simply `mainController.params`. This is easier read and write whilst being just as descriptive.

The [naming section of Effective go](https://golang.org/doc/effective_go.html#names) provides useful guidelines on this topic.

Reply:
Or just export everything in the original struct and do the initialization in main package? That was one of my suggestions in the issue, but I dunno if that's what @dajohi had in mind.

Reply:
There is already a `params` field in MainController struct. I have pushed a new commit.